recipes/aws-sdk-cpp/all/conanfile.py

```diff
@@ -0,0 +1,418 @@
+            "sesv2",
+            "shield",
+            "signer",
+            "sms",
+            "sms-voice",
+            "snowball",
+            "sns",
+            "sqs",
+            "ssm",
+            "sso",
+            "sso-admin",
+            "sso-oidc",
+            "states",
+            "storagegateway",
+            "sts",
+            "support",
+            "swf",
+            "synthetics",
+            "text-to-speech",
+            "textract",
+            "timestream-query",
+            "timestream-write",
+            "transcribe",
+            "transcribestreaming",
+            "transfer",
+            "translate",
+            "waf",
+            "waf-regional",
+            "wafv2",
+            "wellarchitected",
+            "workdocs",
+            "worklink",
+            "workmail",
+            "workmailmessageflow",
+            "workspaces",
+            "xray"
+           )
+    options = {
+            **{ x: [True, False] for x in sdks},
+            **{
+                "shared": [True, False],
+                "fPIC": [True, False],
+                "min_size": [True, False],
+                }
+            }
+    default_options = {key: False for key in options.keys()}
+    default_options["fPIC"] = True
+    default_options["access-management"] = True
+    default_options["identity-management"] = True
+    default_options["queues"] = True
+    default_options["transfer"] = True
+    default_options["s3-encryption"] = True
+    default_options["text-to-speech"] = True
+    internal_requirements = {
+            "access-management": ["iam", "cognito-identity"],
+            "identity-management": ["cognito-identity", "sts"],
+            "queues": ["sqs"],
+            "s3-encryption": ["s3", "kms"],
+            "text-to-speech": ["polly"],
+            "transfer": ["s3"]
+        }
+
+    short_paths = True
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+            if (self.settings.compiler == "gcc"
+                    and tools.Version(self.settings.compiler.version) < "6.0"):
+                raise ConanInvalidConfiguration("""Doesn't support gcc5 / shared.
+                See https://github.com/conan-io/conan-center-index/pull/4401#issuecomment-802631744""")
+
+    def requirements(self):
+        self.requires("aws-c-event-stream/0.1.5")
+        if self.settings.os != "Windows":
+            self.requires("libcurl/7.74.0")
+
+    def package_id(self):
+        for sdk in self.sdks:
+            if getattr(self.options, sdk) and sdk in self.internal_requirements:
+                for internal_requirement in self.internal_requirements[sdk]:
+                    setattr(self.options, internal_requirement, True)
```

Comment:
```suggestion
        for sdk in self.sdks:
            for internal_requirement in self.internal_requirements.get(sdk, []):
                setattr(self.info.options, internal_requirement, True)
```

do not change options values in package_id (or configure), change self.info.options